pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3636,6 +3637,20 @@ private PersistentReplicator getReplicatorReference(String replName, PersistentT
                         p -> new PartitionedTopicMetadata(numPartitions));
                 updatePartition.complete(null);
             } catch (Exception e) {
+                getPartitionedTopicMetadataAsync(topicName, false, false).thenAccept(metadata -> {
+                    int oldPartition = metadata.partitions;
+                    for (int i = oldPartition; i < numPartitions; i++) {
+                        String managedLedgerPath = ZkAdminPaths.managedLedgerPath(topicName.getPartition(i));
+                        try {
+                            namespaceResources().getPartitionedTopicResources().deleteAsync(managedLedgerPath);
+                        } catch (Exception cleanZnodeException) {
+                            log.error("Failed to clean managedLedger znode {}", managedLedgerPath, cleanZnodeException);
+                        }
```

Comment:
`deleteAsync` is an asynchronous method that doesn't throw exception, we shouldn't need to catch the exception whose type we don't know. Instead, we should use `whenComplete` or `exceptionally` to process the exception of the future returned by `deleteAsync`.